plotter/field.go

```diff
@@ -0,0 +1,188 @@
+// Copyright ©2019 The Gonum Authors. All rights reserved.
+// Use of this source code is governed by a BSD-style
+// license that can be found in the LICENSE file.
+
+package plotter
+
+import (
+	"math"
+
+	"gonum.org/v1/plot"
+	"gonum.org/v1/plot/vg"
+	"gonum.org/v1/plot/vg/draw"
+)
+
+// FieldXY describes a two dimensional vector field where the
+// X and Y coordinates are arranged on a rectangular grid.
+type FieldXY interface {
+	// Dims returns the dimensions of the grid.
+	Dims() (c, r int)
+
+	// Vector returns the value of a grid value at (c, r).
```

Comment:
```
	// Vector returns the value of a vector field at (c, r).
	// It will panic if c or r are out of bounds for the field.
```

field.go is derived heavily from heat.go and there are copy/paste errors that reflect that. This is one of those.